packages/test/stochastic-test-utils/src/performActions.ts

```diff
@@ -111,21 +111,28 @@ export async function performFuzzActionsAsync<
 		operation = await generator(state)
 	) {
 		operations.push(operation);
-		if (saveInfo !== undefined && operations.length === saveInfo.saveAt) {
-			await fs.writeFile(saveInfo.filepath, JSON.stringify(operations));
+		if (saveInfo?.saveWhenFinish === false && operations.length === saveInfo.saveAt) {
```

Comment:
Are you making these two options mutually exclusive because they will write to the same file?
Someone stepping through a run whose `SaveInfo` has both the `saveAt` and `saveWhenFinish` turned on, would likely be confused by the fact that the operations are not saved after stepping beyond the iteration that `saveAt` was meant to save.

I think saving twice is less confusing, ideally to different paths.
I see you already save to a modified path for the failure case, maybe it would be best to do that for the success case too.